kumascript/index.js

```diff
@@ -1,79 +1,54 @@
+const { Document } = require("content");
+
+const {
+  INTERACTIVE_EXAMPLES_BASE_URL,
+  LIVE_SAMPLES_BASE_URL,
+} = require("./src/constants");
 const Templates = require("./src/templates.js");
 const AllPagesInfo = require("./src/info.js");
 const { getPrerequisites, render: renderMacros } = require("./src/render.js");
 const {
   getLiveSampleIDs,
   buildLiveSamplePage,
+  LiveSampleError,
 } = require("./src/live-sample.js");
 const { HTMLTool } = require("./src/api/util.js");
 
-class Renderer {
-  constructor({
-    liveSamplesBaseUrl = null,
-    interactiveExamplesBaseUrl = null,
-    uriTransform = (uri) => uri,
-  } = {}) {
-    this.allPagesInfo = null;
-    this.uriTransform = uriTransform;
-    this.liveSamplesBaseUrl = liveSamplesBaseUrl;
-    this.interactiveExamplesBaseUrl = interactiveExamplesBaseUrl;
-    this.templates = new Templates();
-  }
-
-  checkAllPagesInfo() {
-    if (!this.allPagesInfo) {
-      throw new Error(
-        `You haven't yet specified the context for the render via Renderer().use(pageInfoByUri).`
-      );
-    }
-  }
-
-  use(pageInfoByUri) {
-    this.allPagesInfo = new AllPagesInfo(pageInfoByUri, this.uriTransform);
-    return this;
-  }
-
-  clearCache() {
-    this.allPagesInfo.clearCache();
-  }
-
-  async render(source, pageEnvironment, cacheResult = false) {
-    this.checkAllPagesInfo();
-    const uri = pageEnvironment.path.toLowerCase();
-    const cachedResult = this.allPagesInfo.getResultFromCache(uri);
-    if (cachedResult) {
-      return cachedResult;
-    }
-    const [renderedHtml, errors] = await renderMacros(
-      source,
-      this.templates,
-      {
-        ...pageEnvironment,
-        interactive_examples: {
-          base_url: this.interactiveExamplesBaseUrl,
-        },
-        live_samples: { base_url: this.liveSamplesBaseUrl },
+async function renderFromURL(url) {
+  const { rawHtml, metadata } = Document.findByURL(url).document;
+  const [renderedHtml, errors] = await renderMacros(
+    rawHtml,
+    new Templates(),
+    {
+      ...{
+        path: url,
+        url: `${"this.options.sitemapBaseUrl"}${url}`,
```

Comment:
I suppose we should create a new constant (maybe just `SITE_URL`?) that can be overridden from the environment for this? Basically it would be the domain we're deploying to (and its protocol), so for example `https://developer.mozilla.org` or `https://pr9999.yari.io` or `http://localhost:5000` or something like that.

Reply:
Good idea, this was a bit of an easy to overlook placeholder for something that still needs to be worked on, thanks for highlighting it!

Reply:
Out of curiousity: Why do we want to use an absolute URL here? Couldn't we just as well skip the host?